server/executor/trace_poller.go

```diff
@@ -201,3 +206,10 @@ func (tp tracePoller) requeue(job tracePollReq) {
 		tp.enqueueJob(job)
 	}()
 }
+
+func (tp tracePoller) OnPollComplete(callback func(openapi.TestRunResult)) {
+	go func() {
```

Comment:
I guess this is some async programming going on here right?
1. On execution we create a new goroutine.
2. Within the goroutine we wait until the `result` is assigned by the `completePoolChannel` channel.
3. We run the callback with the assigned result value.

Is that right? 😬